arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/Schedule.kt

```diff
@@ -648,9 +648,13 @@ sealed class Schedule<Input, Output> {
      *  For an example see the implementation of [spaced], [linear], [fibonacci] or [exponential]
      */
     @Suppress("UNCHECKED_CAST")
-    fun <A> delayed(delaySchedule: Schedule<A, Duration>): Schedule<A, Duration> =
-      (delaySchedule.modifyDelay { a, b -> a + b } as ScheduleImpl<Any?, A, Duration>)
-        .reconsider { _, dec -> dec.copy(finish = Eval.now(dec.delay)) }
+    fun <A> delayed(delaySchedule: Schedule<A, Duration>): Schedule<A, Duration> {
+      val a = (delaySchedule.modifyDelay { a: Duration, b: Duration -> a + b })
+      val r = (a as? ScheduleImpl<Any?, A, Duration>)
+      return r?.reconsider { _, dec ->
+        dec.copy(finish = Eval.now(dec.delay))
+      }!!
+    }
```

Comment:
Why this change? This can/should never be an incorrect cast, or it's a bug in the Arrow Schedule internals. This cast can never fail due to something on the users side.

Reply:
I was trying to isolate the error. Because the debug point always stops at this position
https://github.com/arrow-kt/arrow-fx/blob/b3cebf1162f22ff92212e1f4cad5c008d6561b2c/arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/Schedule.kt#L444
 and never proceeds to the actual operation `f` here the addtion
https://github.com/arrow-kt/arrow-fx/blob/6e205ab37a4f0d8fe320ad64546dca5913003c75/arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/Schedule.kt#L652